File: doc-examples/example-java/src/main/java/example/gcp/StorageOptionsBuilderCustomizer.java
package example.gcp;

import com.google.cloud.http.HttpTransportOptions;
import com.google.cloud.storage.StorageOptions;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import io.micronaut.core.annotation.NonNull;
import jakarta.inject.Singleton;

//tag::class[]
@Singleton
public class StorageOptionsBuilderCustomizer implements BeanCreatedEventListener<StorageOptions.Builder> {

    @Override
    public StorageOptions.Builder onCreated(@NonNull BeanCreatedEvent<StorageOptions.Builder> event) {
        return event.getBean()
            .setTransportOptions(HttpTransportOptions.newBuilder().setConnectTimeout(60_000).build());
    }
}
//end::class[]

File: doc-examples/example-python/src/main/python/example/gcp/StorageOptionsBuilderCustomizer.py
from com.google.cloud.http import HttpTransportOptions
from com.google.cloud.storage import StorageOptions
from jakarta.inject import Singleton
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener


# tag::class[]
@Singleton
class StorageOptionsBuilderCustomizer(BeanCreatedEventListener[StorageOptions.Builder]):

    def onCreated(self, event: BeanCreatedEvent[StorageOptions.Builder]) -> StorageOptions.Builder:
        return event.getBean() \
            .setTransportOptions(HttpTransportOptions.newBuilder().setConnectTimeout(60_000).build())
# end::class[]
